x/emissions/module/rewards_internal.go

```diff
@@ -0,0 +1,795 @@
+package module
+
+import (
+	"math"
+
+	errors "cosmossdk.io/errors"
+	"github.com/allora-network/allora-chain/x/emissions/types"
+)
+
+// StdDev calculates the standard deviation of a slice of float64.
+// stdDev = sqrt((Σ(x - μ))^2/ N)
+// where μ is mean and N is number of elements
+func StdDev(data []float64) float64 {
+	var mean, sd float64
+	for _, v := range data {
+		mean += v
+	}
+	mean /= float64(len(data))
+	for _, v := range data {
+		sd += math.Pow(v-mean, 2)
+	}
+	sd = math.Sqrt(sd / float64(len(data)))
+	return sd
+}
+
+// flatten converts a double slice of float64 to a single slice of float64
+func flatten(arr [][]float64) []float64 {
+	var flat []float64
+	for _, row := range arr {
+		flat = append(flat, row...)
+	}
+	return flat
+}
+
+// GetWorkerRewardFractions calculates the reward fractions for workers for forecast and inference tasks
+// U_ij / V_ik
+func GetWorkerRewardFractions(scores [][]float64, preward float64) ([]float64, error) {
+	lastScores := make([][]float64, len(scores))
+	for i, workerScores := range scores {
+		end := len(workerScores)
+		start := end - 10
+		if start < 0 {
+			start = 0
+		}
+		lastScores[i] = workerScores[start:end]
+	}
+
+	stdDev := StdDev(flatten(lastScores))
+	var normalizedScores []float64
+	for _, score := range lastScores {
+		normalizedScores = append(normalizedScores, score[len(score)-1]/stdDev)
+	}
+
+	smoothedScores := make([]float64, len(normalizedScores))
+	for i, v := range normalizedScores {
+		res, err := Phi(preward, v)
+		if err != nil {
+			return nil, err
+		}
+		smoothedScores[i] = res
+	}
+
+	total := 0.0
+	for _, score := range smoothedScores {
```

Comment:
the body of this loop could be computed as part of the body of the 2 loops above